CRM/Utils/Rule.php

```diff
@@ -639,12 +639,46 @@ public static function boolean($value) {
    *
    * @return bool
    */
-  public static function email($value) {
+  public static function email($value): bool {
+    if (function_exists('idn_to_ascii')) {
+      $parts = explode('@', $value);
+      foreach ($parts as &$part) {
+        // if the function returns FALSE then let filter_var have at it.
+        $part = self::idnToAsci($part) ?: $part;
+        if ($part === 'localhost' && Civi::settings()->get('environment') === 'Development') {
```

Comment:
Is there a need to check for `Civi::settings()->get('environment') === 'Development'`? Buildkit doesn't set this, the PR test runs and test sites don't set it, I never set it, and the original purpose of that setting I think had to do with stopping cron jobs from accidentally sending things out.

Reply:
@demeritcowboy dunno - I can remove if people think I should - anyone else have an opinion?

Reply:
Emoji poll over on chat https://chat.civicrm.org/civicrm/pl/mckup3magtnid853ysaj5qga3h

Reply:
And the unicorns have it - the environment check is gone

Reply:
Thanks @eileenmcnaughton